app/Http/Controllers/PagesController.php

```diff
@@ -65,41 +65,16 @@ public function help()
     }
 
     /**
-     * @return comments.blade.php
+     * Get links for sorted views based on query
+     * @return popular views
      */
-    protected function comments()
-    {
-        return Project::orderBy('comment_count', 'desc')->paginate(12);
-    }
-
-    /**
-     * @return views.blade.php
-     */
-    protected function views()
-    {
-        return Project::orderBy('views', 'desc')->paginate(12);
-    }
-
-    /**
-     * @return likes.blade.php
-     */
-    protected function likes()
-    {
-        return Project::orderBy('likes', 'desc')->paginate(12);
-    }
-
     public function getLinks()
     {
-        $projects = null;
-        $link = $_GET['popular'];
-
-        if ($link == 'comments') {
-            $projects = $this->comments();
-        } elseif ($link == 'views') {
-            $projects = $this->views();
-        } elseif ($link == 'likes') {
-            $projects = $this->likes();
-        }
+        $q = $_SERVER['QUERY_STRING'];
+        $link = isset($_GET['popular']) ? $_GET['popular'] : 'views';
+        if ($link === 'comments') $link = 'comment_count';
```

Comment:
Can you explain what's going on here?

and why do we have this `$q = $_SERVER['QUERY_STRING'];`